dvc/data/reference.py

```diff
@@ -57,11 +59,16 @@ def to_bytes(self):
         # ReferenceHashFiles should currently only be serialized in
         # memory and not to disk
         fs_path = self.fs_path
+        fs_cls = type(self.fs)
+        mod = None
+        if fs_cls not in FS_MAP.values() and fs_cls != LocalFileSystem:
+            mod = ".".join((fs_cls.__module__, fs_cls.__name__))
         dict_ = {
             self.PARAM_PATH: fs_path,
             self.PARAM_HASH: self.hash_info,
             self.PARAM_CHECKSUM: self.checksum,
             self.PARAM_FS_CONFIG: self.config_tuple(self.fs),
+            self.PARAM_FS_CLS: mod,
```

Comment:
This feels weird to save the cls itself. How about we just save the protocol (e.g. `dvc` for `dvcfs`)?

Reply:
Discussed with @skshetry, turns out that's how fsspec's to_json/from_json behaves too. So keeping it.